browser/ui/views/side_panel/chat_ui/chat_ui_side_panel_coordinator.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/side_panel/chat_ui/chat_ui_side_panel_coordinator.h"
+
+#include <memory>
+
+#include "base/functional/callback.h"
+#include "brave/browser/ui/views/side_panel/playlist/playlist_side_panel_web_view.h"
+#include "brave/browser/ui/webui/chat_ui/chat_ui.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "chrome/app/vector_icons/vector_icons.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_content_proxy.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_entry.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_registry.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_util.h"
+#include "chrome/grit/generated_resources.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/models/image_model.h"
+#include "ui/views/vector_icons.h"
+
+namespace {
+class ChatUISidePanelWebView : public SidePanelWebUIView {
+ public:
+  ChatUISidePanelWebView(Browser* browser,
+                         base::RepeatingClosure close_cb,
+                         BubbleContentsWrapper* contents_wrapper)
+      : SidePanelWebUIView(
+            /* on_show_cb = */ base::RepeatingClosure(),
+            close_cb,
+            contents_wrapper) {}
+  ChatUISidePanelWebView(const ChatUISidePanelWebView&) = delete;
+  ChatUISidePanelWebView& operator=(const ChatUISidePanelWebView&) = delete;
+  ~ChatUISidePanelWebView() override = default;
+};
+}  // namespace
+
+ChatUISidePanelCoordinator::ChatUISidePanelCoordinator(Browser* browser)
+    : BrowserUserData<ChatUISidePanelCoordinator>(*browser) {}
+
+ChatUISidePanelCoordinator::~ChatUISidePanelCoordinator() = default;
+
+void ChatUISidePanelCoordinator::CreateAndRegisterEntry(
+    SidePanelRegistry* global_registry) {
+  std::u16string title = u"ChatUI";
+
+  global_registry->Register(std::make_unique<SidePanelEntry>(
+      SidePanelEntry::Id::kChatUI, title, ui::ImageModel(),
+      base::BindRepeating(&ChatUISidePanelCoordinator::CreateWebView,
+                          base::Unretained(this))));
+}
+
+std::unique_ptr<views::View> ChatUISidePanelCoordinator::CreateWebView() {
+  const bool should_create_contents_wrapper = !contents_wrapper_;
+  if (should_create_contents_wrapper) {
+    contents_wrapper_ = std::make_unique<BubbleContentsWrapperT<ChatUI>>(
+        GURL(kChatUIURL), GetBrowser().profile(),
+        IDS_SIDEBAR_PLAYLIST_ITEM_TITLE,
+        /*webui_resizes_host=*/false,
+        /*esc_closes_ui=*/false);
+    contents_wrapper_->ReloadWebContents();
```

Comment:
Why we need reload here?

Reply:
If we don't reload here the WebUI doesn't render. It's similar code to Playlist as well.